tpdatasrc/tpgamefiles/scr/tpModifiers/sp_align_weapon.py

```diff
@@ -0,0 +1,78 @@
+from templeplus.pymod import PythonModifier
+from toee import *
+import tpdp
+from utilities import *
+import spell_utils
+print "Registering sp-Align Weapon"
+
+def alignWeaponSpellOnConditionAdd(attachee, args, evt_obj):
+    spellPacket = tpdp.SpellPacket(args.get_arg(0))
+    spellPacket.add_target(attachee, 0)
+    attachee.item_condition_add_with_args('Weapon Align', 0, args.get_arg(2), 0, 0, args.get_arg(0))
+    return 0
+
+def alignWeaponSpellOnConditionRemove(attachee, args, evt_obj):
+    attachee.item_condition_remove('Weapon Align', args.get_arg(0))
+    return 0
+
+alignWeaponSpell = PythonModifier("sp-Align Weapon", 4) # spell_id, duration, alignType, empty
+alignWeaponSpell.AddHook(ET_OnConditionAdd, EK_NONE, alignWeaponSpellOnConditionAdd,())
+alignWeaponSpell.AddHook(ET_OnConditionRemove, EK_NONE, alignWeaponSpellOnConditionRemove, ())
+alignWeaponSpell.AddHook(ET_OnD20Query, EK_Q_Critter_Has_Spell_Active, spell_utils.queryActiveSpell, ())
+alignWeaponSpell.AddHook(ET_OnD20Signal, EK_S_Killed, spell_utils.spellKilled, ())
+alignWeaponSpell.AddSpellDispelCheckStandard()
+alignWeaponSpell.AddSpellTeleportPrepareStandard()
+alignWeaponSpell.AddSpellTeleportReconnectStandard()
+alignWeaponSpell.AddSpellCountdownStandardHook()
+
+
+###### Align Weapon Condition ######
+
+def weaponAlignOnDamage(attachee, args, evt_obj):
+    usedWeapon = evt_obj.attack_packet.get_weapon_used()
+    if not spell_utils.verifyItem(usedWeapon, args):
+        return 0
+    #Check if weapon is already aligned
+    if evt_obj.damage_packet.attack_power & D20DAP_HOLY:
+        return 0
+    elif evt_obj.damage_packet.attack_power & D20DAP_UNHOLY:
+        return 0
+    elif evt_obj.damage_packet.attack_power & D20DAP_LAW:
+        return 0
+    elif evt_obj.damage_packet.attack_power & D20DAP_CHAOS:
+        return 0
+
+    alignType = args.get_arg(1)
+
+    if alignType == 1:
```

Comment:
IMHO there should be particle effect only when it's really necessary, otherwise it will wear out and start to annoy each time. Essentially aligned weapon should bypass aligned damage reduction. For example Hezrou suppose to have Damage reduction 10/good. So when alignment of a weapon does no real bonus, then there should be no particle.

Reply:
Will look into it